2014/B_U1/main.cpp

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+using namespace std;
+
+struct Uzsakovas
+{
+    int x;
+    int y;
+    int atstumasNuoPicerijosIrAtgal()
+    {
+        return (abs(x) + abs(y)) * 2;
+    }
+};
+
+void skaitytiIsFailo(int &kilometrazas, vector<Uzsakovas> &uzsakovai)
+{
+    Uzsakovas laikinasUzsakovas;
+    int uzsakovuKiekis;
+    ifstream data("U1.txt");
+
+    data >> uzsakovuKiekis >> kilometrazas;
+
+    for (int i = 0; i < uzsakovuKiekis; i++)
+    {
+        data >> laikinasUzsakovas.x >> laikinasUzsakovas.y;
+        uzsakovai.push_back(laikinasUzsakovas);
+    }
+    data.close();
+}
+
+void skaiciuotiKiekAptarnavoIrNuvaziavo(int &aptarnautuKiekis, int &nuvaziavo, int kilometrazas, vector<Uzsakovas> uzsakovai)
+{
+    for (int i = 0; i < uzsakovai.size(); i++)
+    {
+        if (uzsakovai[i].atstumasNuoPicerijosIrAtgal() + nuvaziavo > kilometrazas)
+        {
+            nuvaziavo += uzsakovai[i].atstumasNuoPicerijosIrAtgal();
+            aptarnautuKiekis++;
+            break;
+        }
+        nuvaziavo += uzsakovai[i].atstumasNuoPicerijosIrAtgal();
+        aptarnautuKiekis++;
```

Comment:
No need to do accumulation twice. Check can be done in the end of a loop after adding up the numbers.
No need to call `uzsakovai[i].atstumasNuoPicerijosIrAtgal()` twice.